src/co/elastic/NotificationManager.groovy

```diff
@@ -367,19 +373,57 @@ def generateBuildReport(Map args = [:]) {
     return output
 }
 
-def queryFilter(timeout, flakyThreshold) {
-  return """{
-                "timeout": "${timeout}",
-                "sort" : [
-                  { "timestamp" : "desc" },
-                  { "test_score" : "desc" }
-                ],
-                "query" : {
-                  "range" : {
-                    "test_score" : {
-                      "gt" : ${flakyThreshold}
-                    }
-                  }
+def queryFilter(timeout) {
+  return """
+{
+  "aggs": {
+    "test_name": {
+      "terms": {
+        "field": "test.name.keyword",
+        "order": {
+          "_count": "desc"
+        }
+      }
+    }
+  },
+  "size": 0,
+  "script_fields": {},
+  "stored_fields": [
+    "*"
+  ],
+  "runtime_mappings": {},
+  "_source": {
+    "excludes": []
+  },
+  "query": {
+    "bool": {
+      "must": [],
+      "filter": [
+        {
+          "bool": {
+            "should": [
+              {
+                "match_phrase": {
+                  "job.fullName": "Beats/beats/master"
```

Comment:
Maybe the env variable JOB_NAME could help here:

>JOB_NAME
Name of the project of this build, such as "foo" or "foo/bar".


For instance:

```
$ curl -s 'https://beats-ci.elastic.co/job/Beats/job/beats/job/master/lastSuccessfulBuild/artifact/env-info.json' | jq .JOB_NAME
"Beats/beats/master"
```

Although IIRC, the flaky test analyser only worked for the master branch. so if working with a PR I'd say it should then be forced to the master branch

Reply:
We faced a similar problem in the past which is how we ended up with [flakyReportIdx](https://github.com/search?p=1&q=org%3Aelastic+flakyReportIdx&type=Code), which allowed individual pipelines to select the index that they wanted the Notification Manager class to do their Elasticsearch lookup against.

It's a bit disruptive, but what I think we should do here is to immediately deprecate `flakyReportIdx` as an option and raise a warning if it is passed in. Then we should add a new option called `jobName`. Consumers of the step would typically pass in the `JOB_NAME` variable as you suggest, but in certain cases (such as Beats), [this could be modified](https://github.com/elastic/beats/blob/024dd5c98021368bda3c7b8b0a91f69c9b14d5aa/Jenkinsfile#L169-L176). 

@v1v WDYT?

Reply:
I have updated this to do the following:

1. Log a warning and exit early if just `flakyReportIdx` is passed in.
2. Error if both `flakyReportIdx` and `jobName` are used.
3. Error if neither `flakyReportIdx` nor `jobName` are used.

Reply:
I have removed this check per https://github.com/elastic/apm-pipeline-library/pull/1016#discussion_r590522960